### PR TITLE
Clean up wording in releases.md

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: docs/welcome/releases.md [hunk 1]
```diff
@@ -31,15 +25,16 @@ Kubernetes version.
 
 
 Use the latest version of kops for all releases of Kubernetes, with the caveat
-that higher versions of Kubernetes are not _officially_ supported by kops. Releases who are ~~crossed out~~ _should_ work but we suggest should be upgraded soon.
+that higher versions of Kubernetes are not _officially_ supported by kops.
+Releases who are ~~crossed out~~ _should_ work, but we suggest they be upgraded soon.
```
Comment (MoShitrit) [inline docs/welcome/releases.md:29]: Would "Releases _which_ are..." suit better here?

File: docs/welcome/releases.md [hunk 2]
```diff
@@ -1,23 +1,17 @@
...
+do our best to support previous releases for some period.
 
-One exception, in regard to compatibility, kops supports the equivalent
-Kubernetes minor release number.  A minor version is the second digit in the
-release number.  kops version 1.16.0 has a minor version of 16. The numbering
-follows the semantic versioning specification, MAJOR.MINOR.PATCH.
+Kops does not, however, support Kubernetes releases with a greater minor release number
+than that of kops. A minor version is the second digit in the
+release number: kops version 1.16.0 has a minor version of 16. The numbering
+follows the semantic versioning specification: MAJOR.MINOR.PATCH.
```
Comment (gjtempleton) [inline docs/welcome/releases.md:11]: Worth linking out to the semver spec here?

Comment (johngmyers) [reply docs/welcome/releases.md:11]: The semver spec would be highly parenthetical. The key point here is that it is the second part that counts.

Though now I notice it's not the second _digit_ and, pedantically, the major version number also matters.

**Pull request merged**